libsolidity/experimental/analysis/TypeClassRegistration.cpp

```diff
@@ -42,10 +42,8 @@ bool TypeClassRegistration::analyze(SourceUnit const& _sourceUnit)
 
 bool TypeClassRegistration::visit(TypeClassDefinition const& _typeClassDefinition)
 {
-	Type typeVar = m_typeSystem.freshTypeVariable({});
-
 	std::variant<TypeClass, std::string> typeClassOrError = m_typeSystem.declareTypeClass(
-		typeVar,
+		m_typeSystem.freshTypeVariable({}),
```

Comment:
It's the only thing evaluated as parameter so there's no problem here.

But ok, I did no realize that you did this intentionally. You have a point that this could turn into a problem later if other calls get added in parameters. I'm going to revert this change.

Reply:
It's the only thing, until it isn't :-). It intentionally creates anything that may depend on evaluation order up front to establish that as pattern to avoid running into issues due to this eventually.

Reply:
Yes. I know. And I admit I was not paying enough attention to that. So I'll remove it, but in a way, it's good I did this, because it made you comment and reminded me I should be more careful about this :)